src/ManagedArray.inl

```diff
@@ -117,7 +119,7 @@ CHAI_HOST_DEVICE ManagedArray<T>::ManagedArray(ManagedArray const& other):
    */
   if (!std::is_const<T>::value) {
     CHAI_LOG("ManagedArray", "T is non-const, registering touch of pointer" << m_active_pointer);
-    T_non_const* non_const_pointer = const_cast<T_non_const*>(m_active_pointer);
+//    T_non_const* non_const_pointer = const_cast<T_non_const*>(m_active_pointer);
```

Comment:
@davidbeckingsale this was a compilation warning for unused variable. is it needed for something?